cmd/controller-manager/app/controller-manager.go

```diff
@@ -339,9 +274,22 @@ func setOptionsByKubeFedConfig(opts *options.Options) {
 				Namespace: qualifiedName.Namespace,
 			},
 		}
+
+		setDefaultKubeFedConfigScope(fedConfig)
+		createKubeFedConfig(opts.Config.KubeConfig, fedConfig)
 	}
 
-	SetDefaultKubeFedConfig(fedConfig)
+	qualifedName := util.QualifiedName{
+		Name:      fedConfig.Name,
+		Namespace: fedConfig.Namespace,
+	}
+
+	errs := validation.ValidateKubeFedConfig(fedConfig)
```

Comment:
I'm a bit confused as to why it's desirable to validate in this particular location. Would a kubefed api created on line 279 or loaded from the API need validation, or is it just the case of when a config was provided via a yaml file? 

(No action required) Does this imply that any of the controllers that can call validation in code should do so rather than making a call to the webhook?